### PR TITLE
Allow passing 'description' 'ManagedZone.__init__'

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1728 as a base.~~

Likewise, allow passing 'description' to 'Client.zone'.

Closes: #1721

### CODE REVIEW

File: gcloud/dns/zone.py [hunk 1]
```diff
@@ -220,9 +228,11 @@ def _build_resource(self):
         """Generate a resource for ``create`` or ``update``."""
         resource = {
             'name': self.name,
-            'dnsName': self.dns_name,
         }
 
+        if self.dns_name is not None:
+            resource['dnsName'] = self.dns_name
```
Comment (dhermes) [inline gcloud/dns/zone.py]: This change seems to belong in another PR?

Comment (tseaver) [inline gcloud/dns/zone.py]: Maybe it should've been in #1728, which allowed `dns_name` to be None;  the coverage gap didn't surface until here (because `dns_name` is now (as of this PR) the fallback for `description`, and because it is now allowed to be `None`).

Comment (tseaver) [inline gcloud/dns/zone.py]: This is actually a bit of a weird case:  if the user doesn't pass `dns_name` or `description` to `ManagedZone.__init__`, and doesn't assign them, the API call in `ManagedZone.create` is going to fail anyway:  our usual rule is to let the back-end enforce those rules, so I didn't attempt any validation here:  I'm merely minimizing the resource sent with the `create` request (as the back-ends seem to prefer, at least based on how they return them).

Comment (dhermes) [inline gcloud/dns/zone.py]: OK SGTM.

**Comment by dhermes** [comment]
LGTM
